src/components/structures/RoomView.js

```diff
@@ -1281,7 +1281,7 @@ module.exports = React.createClass({
         // we want to show the bar if the read-marker is off the top of the
         // screen.
         // If pos is null, the event might not be paginated, so show the unread bar!
-        var showBar = pos < 0 || pos === null;
+        var showBar = (pos < 0 || pos === null) && this.refs.messagePanel.isReadMarkerSet();
```

Comment:
It's used in a few other places actually. We could have `shouldShowJumpToRM` on `TimelinePanel`?